library-udf/src/main/java/org/apache/iotdb/library/util/Util.java

```diff
@@ -0,0 +1,302 @@
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iotdb.library.util;
+
+import org.apache.iotdb.db.query.udf.api.access.Row;
+import org.apache.iotdb.db.query.udf.api.collector.PointCollector;
+import org.apache.iotdb.tsfile.file.metadata.enums.TSDataType;
+
+import org.apache.commons.math3.stat.descriptive.rank.Median;
+import org.eclipse.collections.api.tuple.primitive.LongIntPair;
+import org.eclipse.collections.impl.map.mutable.primitive.LongIntHashMap;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+/** This class offers functions of getting and putting values from iotdb interface. */
+public class Util {
+
+  /**
+   * Get value from specific column from Row, and cast to double. Make sure never get null from Row.
+   *
+   * @param row data row
+   * @param index the column index
+   * @return value of specific column from Row
+   * @throws NoNumberException when getting a no number datatype
+   */
+  public static double getValueAsDouble(Row row, int index) throws NoNumberException, IOException {
+    double ans = 0;
+    switch (row.getDataType(index)) {
+      case INT32:
+        ans = row.getInt(index);
+        break;
+      case INT64:
+        ans = row.getLong(index);
+        break;
+      case FLOAT:
+        ans = row.getFloat(index);
+        break;
+      case DOUBLE:
+        ans = row.getDouble(index);
+        break;
+      default:
+        throw new NoNumberException();
+    }
+    return ans;
+  }
+
+  /**
+   * Get value from 0th column from Row, and cast to double. Make sure never get null from Row.
+   *
+   * @param row data row
+   * @return value from 0th column from Row
+   * @throws NoNumberException when getting a no number datatype
+   */
+  public static double getValueAsDouble(Row row) throws NoNumberException, IOException {
+    return getValueAsDouble(row, 0);
+  }
+
+  /**
+   * Get value from 0th column from Row, and cast to Object.
+   *
+   * @param row data row
+   * @return value from 0th column from Row
+   */
+  public static Object getValueAsObject(Row row) throws IOException {
+    Object ans = 0;
+    switch (row.getDataType(0)) {
+      case INT32:
+        ans = row.getInt(0);
+        break;
+      case INT64:
+        ans = row.getLong(0);
+        break;
+      case FLOAT:
+        ans = row.getFloat(0);
+        break;
+      case DOUBLE:
+        ans = row.getDouble(0);
+        break;
+      case BOOLEAN:
+        ans = row.getBoolean(0);
+        break;
+      case TEXT:
+        ans = row.getString(0);
+        break;
+    }
+    return ans;
```

Comment:
Whether to return -1 if the case does not meet the requirement

Reply:
It already covers all data types. Perhaps throw an error when none is satisfied?